### PR TITLE
Fix 500 error when adding host with tags in string format (RHCLOUD-3342)

### PR DESCRIPTION
**Pull request by ryandillinfelton**
Fixes 500 error when sending string formatted tag instead of structured tag while creating host. Now returns 400

### CODE REVIEW

File: swagger/api.spec.yaml [hunk 1]
```diff
@@ -729,9 +731,12 @@ components:
           items:
             $ref: '#/components/schemas/FactSet'
         tags:
+          description: An array of the tags on the host
           type: array
           items:
-            $ref: '#/components/schemas/StructuredTag'
+            oneOf:
+              - $ref: '#/components/schemas/StructuredTag'
```
Comment (jharting) [inline swagger/api.spec.yaml]: We're getting a 500 because of the mismatch of validation constraints on the REST layer. The output format is stricter (only allows structured tag representation) than the input format (allows tags field of any format). As a result, an invalid representation can get in but cannot get out (does not pass output validation).

Instead of relaxing the output validation (by adding MarshmallowError) we should make input validation more strict (validate the tags field)

**Comment by ryandillinfelton** [comment]
Re-review request was an accident

File: swagger/api.spec.yaml (hunk 1 above)
Comment (ryandillinfelton) [reply swagger/api.spec.yaml]: Is there somewhere else I ought to be doing input validation here? I have Marshmallow validation for tags which seems to be catching tags that are in the wrong format, but breaking the response schema with the error response which is why I thought relaxing the response was the way to go.

I have noticed that some of the validation errors that marshmallow creates for tags create a proper response without relaxing the output validation which is strange, I haven't been able to pin down why though,

Comment (ryandillinfelton) [reply swagger/api.spec.yaml]: Actually, i think I've figured it out now

File: swagger/api.spec.yaml [hunk 2]
```diff
@@ -729,9 +734,10 @@ components:
           items:
             $ref: '#/components/schemas/FactSet'
         tags:
+          description: An array of the tags on the host
           type: array
           items:
-            $ref: '#/components/schemas/StructuredTag'
+            - $ref: '#/components/schemas/StructuredTag'
```
Comment (jharting) [inline swagger/api.spec.yaml]: Why this change? It does not look correct. (https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#response-object-examples) Other than this the PR looks good.

**Comment by jharting** [comment]
Also please remember to put JIRA id to PR and commit messages.

File: swagger/api.spec.yaml (hunk 2 above)
Comment (ryandillinfelton) [reply swagger/api.spec.yaml]: Ahh, I missed removing the dash when I removed the "oneOf" part and the marshmallow error schema
